Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,282 @@
+/-
+Copyright (c) 2024 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan
+-/
+import Mathlib.FieldTheory.KrullTopology
+import Mathlib.FieldTheory.Galois.GaloisClosure
+import Mathlib.Topology.Algebra.ClosedSubgroup
+/-!
+
+# The Fundamental Theorem of Infinite Galois Theory
+
+In this file, we proved the fundamental theorem of infinite Galois theory and the special case for
+open subgroups and normal subgroups. We first verify that IntermediateField.fixingSubgroup and
+IntermediateField.fixedField are inverses of each other between IntermediateFields and
+closed subgroups of the Galois group.
+
+# Main definitions and results
+
+In `K/k`, for any intermediate field `L` :
+
+* `fixingSubgroup_isClosed` : the fixing subgroup of `L`
+  (`Gal(K/L)`) is closed.
+
+* `fixedField_fixingSubgroup` : the fixing field of the
+  fixing subgroup of `L` is equal to `L` itself.
+
+For any subgroup of `Gal(K/k)` `H` :
+
+* `restrict_fixedField` : For a Galois intermediate field `M`, the fixed field of the image of `H`
+  restricted to `M` is equal to the fixed field of `H` intersected with `M`.
+* `fixingSubgroup_fixedField` : If `H` is closed, the fixing subgroup of the fixed field of `H`
+  is equal to `H` itself.
+
+The fundamental theorem of infinite galois theory :
+
+* `IntermediateFieldEquivClosedSubgroup` : The order equivalence is given by mapping any
+  intermediate field `L` to the fixing subgroup of `L`, with its inverse mapping any
+  closed subgroup of `Gal(K/k)` `H` to the fixed field of `H`. The composition is equal to
+  the identity as described in the lemmas above, and compatibility with the order follows easily.
+
+Special cases :
+
+* `OpeniffFinite` : The fixing subgroup of an intermediate field `L` is open iff
+  `L` is finite-dimensional.
+
+* `NormaliffGalois` : The fixing subgroup of an intermediate field `L` is normal iff
+  `L` is Galois.
+
+-/
+
+variable {k K : Type*} [Field k] [Field K] [Algebra k K]
+
+namespace InfiniteGalois
+
+open Pointwise FiniteGaloisIntermediateField AlgEquiv
+--Note: The `adjoin`s below are `FiniteGaloisIntermediateField.adjoin`
+
+instance : TopologicalSpace (K ≃ₐ[k] K) := inferInstance
```

Comment:
If typeclass inference already knows this, then is this line necessary? If it is, can you add a comment explaining why?

Reply:
It could be deleted, it is only somehow needed while proving, nevermind.